web/modules/custom/dept_migrate/src/MigrateUuidLookupManager.php

```diff
@@ -138,6 +139,69 @@ public function lookupBySourceNodeId(array $nids) {
     return $map;
   }
 
+  /**
+   * @param array $fids
+   *   One or more source file ids.
+   *
+   * @return array
+   *   Associative array of file metadata, keyed by source file id
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function lookupBySourceFileId(array $fids) {
```

Comment:
This is a bit rough around the edges, but seems to work for files, images and video entity types. Will adapt/tidy as needed